src/router_core/connections.c

```diff
@@ -1329,6 +1329,9 @@ static void qdr_connection_closed_CT(qdr_core_t *core, qdr_action_t *action, boo
     while (link_ref) {
         qdr_link_t *link = link_ref->link;
 
+        if (link->auto_link && link->auto_link->retry_timer)
+            qdr_core_timer_cancel_CT(core, link->auto_link->retry_timer);
```

Comment:
The fact that there is a retry-timer in the auto-link is an implementation detail that should not be visible at this level.  It would be better to have a call in the route_control module for "auto_link_connection_closed" or similar.